tests/regionsearch/test_indexer.py

```diff
@@ -0,0 +1,60 @@
+"""Tests for indexer classes."""
+
+import astropy.units as u
+import numpy
+
+from kbmod.regionsearch.indexers import PartitionIndexer
+from kbmod.regionsearch.utilities import RegionSearchClusterData
+
+
+def test_partition_indexer():
+    """Test the PartitionIndexer"""
+    data = RegionSearchClusterData(clustercnt=2, samplespercluster=5, removecache=True)
+    clusteri = 0
+    indexer = PartitionIndexer(
+        search_ra=(data.clusters[clusteri][0]) * u.deg,
+        search_dec=(data.clusters[clusteri][1]) * u.deg,
+        search_distance=data.clusterdistances[clusteri] * u.au,
+        search_fov=2.0 * u.deg,
+        is_in_index=clusteri,
+        is_out_index=~clusteri,
+    )
+    assert indexer is not None
+    indices = indexer.observations_to_indices(
+        pointing=data.observation_pointing,
+        time=data.observation_time,
+        fov=2.0 * u.deg,
+        location=data.observation_geolocation,
+    )
+    assert indices is not None
+    indexref = numpy.where(numpy.array(data.cluster_id) == clusteri, clusteri, ~clusteri)
+    assert numpy.all(indexref == indices)
+
+
+def test_cache_write_read():
+    """Test the PartitionIndexer"""
+    data1 = RegionSearchClusterData(samplespercluster=5, removecache=True)
+    assert data1 is not None
+    assert data1.data_loaded
+    assert data1.data_generated == True
+    # check that written files can be read
+    data2 = RegionSearchClusterData(samplespercluster=5, removecache=False)
+    assert data2 is not None
+    assert data2.data_loaded
+    assert data2.data_generated == False
+    assert all(data1.observation_pointing == data2.observation_pointing)
```

Comment:
This is check checking that when you generate and then read the data you are getting the same result. 

You should also include a test that checks that you are getting reasonable results from the data generation. Specifically that the cluster IDs contain the occurrence of each of the 5 clusters.

Reply:
added a test for that. Not sure it is the test you intended though.